### PR TITLE
SOC-3194 Move trending/latest to sidebar

### PR DESCRIPTION
**Pull request by latata**
## Links

* https://wikia-inc.atlassian.net/browse/SOC-3194

## Description

Sorting moved to left filter sidebar

## Reviewers
@Wikia/social-frontend

### CODE REVIEW

File: front/main/app/components/discussion-sort.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import Ember from 'ember';
+import {track, trackActions} from '../utils/discussion-tracker';
+
+export default Ember.Component.extend(
+	{
+		classNames: ['discussion-sort'],
+		classNameBindings: ['noTrending'],
+		discussionSort: Ember.inject.service(),
+		sortBy: Ember.computed('discussionSort.sortBy', function () {
+			if (this.get('disabled')) {
```
Comment (mleczey) [inline front/main/app/components/discussion-sort.js]: This could be probably one liner.

File: front/main/app/components/discussion-sort.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import Ember from 'ember';
+import {track, trackActions} from '../utils/discussion-tracker';
+
+export default Ember.Component.extend(
+	{
+		classNames: ['discussion-sort'],
+		classNameBindings: ['noTrending'],
```
Comment (mleczey) [inline front/main/app/components/discussion-sort.js]: Is this needed here?

File: front/main/app/styles/component/discussion/_discussion-sort.scss [hunk 3]
```diff
@@ -0,0 +1,40 @@
+.discussion-sort {
+	ul {
```
Comment (mleczey) [inline front/main/app/styles/component/discussion/_discussion-sort.scss]: I think it would be better to relate not on `ul` but on css class.

File: front/main/app/styles/component/discussion/_discussion-sort.scss [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+		margin-left: 0;
+
+		li {
+			clear: left;
+			color: $discussion-filters-color;
+			list-style: none;
+			width: 100%;
+
+			label {
```
Comment (mleczey) [inline front/main/app/styles/component/discussion/_discussion-sort.scss]: I think it would be better to relate not on `label` but on css class.

File: front/main/app/styles/component/discussion/_discussion-sort.scss [hunk 5]
```diff
@@ -0,0 +1,40 @@
+.discussion-sort {
+	ul {
+		margin-left: 0;
+
+		li {
+			clear: left;
```
Comment (mleczey) [inline front/main/app/styles/component/discussion/_discussion-sort.scss]: Could this be replaced with flex box? (Convention here is to use flexbox instead of float.)

File: front/main/app/styles/mixin/radio-fancy.scss [hunk 6]
```diff
@@ -0,0 +1,24 @@
+@mixin radio-fancy($size) {
```
Comment (mleczey) [inline front/main/app/styles/mixin/radio-fancy.scss]: Is this discussions related? If yes maybe you could move that to discussions mixins.

File: front/main/app/templates/components/discussion-sort.hbs [hunk 7]
```diff
@@ -0,0 +1,16 @@
+<h3 class="discussion-filter-header">
+	{{i18n 'main.sort-by-header' ns='discussion'}}
+</h3>
+<fieldset class="discussion-fieldset sortby-fieldset" disabled={{disabled}}>
+	<ul class="sort-options">
+		{{#each discussionSort.sortTypes as |sortType|}}
+			<li class="sort-{{sortType.name}}">
+				<input type="radio" value="{{sortType.name}}" {{action 'setSortBy' sortType.name}} name="sort-button" class="bullet" id="sort-button-{{sortType.messageKey}}">
```
Comment (mleczey) [inline front/main/app/templates/components/discussion-sort.hbs]: Should `name` here be unique?

File: front/main/app/templates/components/discussion-reported-filter.hbs [hunk 8]
```diff
@@ -1,13 +1,15 @@
...
-<div class="collapsable-container">
-	{{wikia-ui-components.fancy-check-box
-		inputId='discussion-reported-filter'
-		class='moderation-checkbox-label'
-		isChecked=onlyReported
-		label=(i18n 'main.reported-content-filter-text' ns='discussion')
-		update=(action 'toggleReported')
-	}}
-</div>
+<fieldset class="discussion-fieldset moderation-fieldset">
```
Comment (mleczey) [inline front/main/app/templates/components/discussion-reported-filter.hbs]: Why did you move classes and tag name from `.js` file?

File: front/main/app/templates/components/discussion-entities-list-layout.hbs [hunk 9]
```diff
@@ -58,8 +58,10 @@
 			{{#if canShowReportedPostsFilter}}
 				{{discussion-reported-filter
 					applyFilters=(action this.attrs.applyFilters)
+					topDecoration=reportedFilterTopDecoration
 				}}
 			{{/if}}
+			{{discussion-sort setSortBy=(action 'setSortBy') disabled=isSortingDisabled}}
```
Comment (mleczey) [inline front/main/app/templates/components/discussion-entities-list-layout.hbs]: Please split each assignment to different line.

File: front/main/app/components/discussion-sort.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+			setSortBy(sortBy) {
+				if (sortBy !== this.get('discussionSort.sortBy')) {
+					if (sortBy === 'latest') {
+						track(trackActions.LatestPostTapped);
+					} else if (sortBy === 'trending') {
+						track(trackActions.TrendingPostTapped);
+					}
+				}
+
+				this.get('setSortBy')(sortBy);
```
Comment (mleczey) [inline front/main/app/components/discussion-sort.js]: Should this be called always? Event when sort by does not change? If there is no such situation, why check in first line?

File: front/main/app/templates/components/discussion-reported-filter.hbs (hunk 8 above)
Comment (latata) [inline front/main/app/templates/components/discussion-reported-filter.hbs]: I needed wrapper div for margin and border
